src/app/submission/pages/submission.component.ts

```diff
@@ -277,31 +319,44 @@ export class SubmissionComponent implements OnInit, OnDestroy {
     return links && links[linkName] ? links[linkName]['href'] : null;
   }
 
-  downloadFile() {
-    this.downloadDisabled = true;
+  generateSpreadsheet() {
+    const uuid = this.submissionEnvelope['uuid']['uuid'];
+    this.brokerService.generateSpreadsheetFromSubmission(uuid)
+      .subscribe(response => {
+          this.alertService.success('',
+            'Successfully triggered spreadsheet generation! This may take a while. ' +
+            'Please come back later and check the link to download the file.');
+          this.downloadJobOngoing = true;
+        },
+        error => {
+          const err = 'An error occurred in the request to generated the spreadsheet'
+          this.alertService.error('Error', err);
+          console.error(err, error);
+        });
+  }
+
+  downloadSpreadsheet() {
     const uuid = this.submissionEnvelope['uuid']['uuid'];
     this.loaderService.display(true, 'This may take a moment. Please wait...');
-    // now set a cookie to disable further download attempt within X min of this request or
-    // until a response is returned when the cookie gets deleted.
-    this.cookieService.set(this.DOWNLOAD_TIMEOUT_COOKIE, "1", {expires: this.DOWNLOAD_BACKOFF_MINS / (24*60)})
+
     this.brokerService.downloadSpreadsheet(uuid).subscribe(response => {
-      const filename = response['filename'];
-      const newBlob = new Blob([response['data']]);
+        const filename = response['filename'];
+        const newBlob = new Blob([response['data']]);
         this.saveFileService.saveFile(newBlob, filename);
         this.loaderService.display(false);
-      this.downloadDisabled = false;
-      this.cookieService.delete(this.DOWNLOAD_TIMEOUT_COOKIE);
-    },
-    err => {
-      if (err instanceof TimeoutError) {
+      },
+      err => {
+        const retry_message = 'Please retry later.';
+        if (err instanceof TimeoutError) {
+          this.alertService.error('', 'Spreadsheet download timed out. ' + retry_message, false, true);
```

Comment:
What is the empty string parameter's purpose?

Reply:
The first param is the title. I didn't think much about the error message. I'll let gabs/wrangler suggest proper error messages/titles. I just added "Success"/"Error" for now. 

<img width="1044" alt="Screenshot 2022-01-31 at 19 56 50" src="https://user-images.githubusercontent.com/31281398/151863504-ac7a4dfe-3323-4703-96cc-9e271c8a2af5.png">